tests/integration/test_schema_protobuf.py

```diff
@@ -1270,3 +1270,106 @@ async def test_protobuf_update_ordering(registry_async_client: Client) -> None:
     assert res.status_code == 200
     assert "id" in res.json()
     assert schema_id != res.json()["id"]
+
+
+async def test_protobuf_normalization_of_options(registry_async_client: Client) -> None:
+    subject = create_subject_name_factory("test_protobuf_normalization")()
+
+    schema_with_option_unordered_1 = """\
+syntax = "proto3";
+package tc4;
+
+option java_package = "com.example";
+option java_outer_classname = "FredProto";
+option java_multiple_files = true;
+option java_generic_services = true;
+option java_generate_equals_and_hash = true;
+option java_string_check_utf8 = true;
+
+message Foo {
+  string code = 1;
+}
+"""
+
+    body = {"schemaType": "PROTOBUF", "schema": schema_with_option_unordered_1}
+    res = await registry_async_client.post(f"subjects/{subject}/versions?normalize=true", json=body)
+
+    assert res.status_code == 200
+    assert "id" in res.json()
+    original_schema_id = res.json()["id"]
+
+    schema_with_option_unordered_2 = """\
+syntax = "proto3";
+package tc4;
+
+option java_package = "com.example";
+option java_generate_equals_and_hash = true;
+option java_string_check_utf8 = true;
+option java_multiple_files = true;
+option java_outer_classname = "FredProto";
+option java_generic_services = true;
+
+message Foo {
+  string code = 1;
+}
+"""
+
+    body = {"schemaType": "PROTOBUF", "schema": schema_with_option_unordered_2}
+    res = await registry_async_client.post(f"subjects/{subject}", json=body)
+    assert res.status_code == 404
+
+    res = await registry_async_client.post(f"subjects/{subject}?normalize=true", json=body)
+
+    assert res.status_code == 200
+    assert "id" in res.json()
+    assert original_schema_id == res.json()["id"]
+
+
+async def test_protobuf_normalization_of_options_specify_version(registry_async_client: Client) -> None:
```

Comment:
I failed to see what was beef in this second test compared to first? There seems to be same schemas used so maybe just one test with more API calls, and/or some comments about intention?

Also integration test could fetch the schema by id after registering it with `normalize=true` to verify that fetch returns normalized version.

Reply:
The `test_protobuf_normalization_of_options` was verifying that producing a schema with the normalized does correctly register a normalized schema because by trying to seek for the schema later without normalizing it we didn't find it while by normalizing it we can and so its registered. While the `test_protobuf_normalization_of_options_specify_version` verify that its idempotent and the second time we try to register the same normalized schema the returned id its the same, so the schema its not registered twice. Probably the second one isn't really specific to the `normalize` option and can be transformed into verify that the returned schema its a normalized schema